iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/helplets.inl

```diff
@@ -64,23 +66,20 @@ inline bool isValidFileName(const string<StringCapacity>& name) noexcept
 template <uint64_t StringCapacity>
 inline bool isValidFilePath(const string<StringCapacity>& name) noexcept
 {
-    if (name.empty())
+    if (doesEndWithPathSeparator(name))
```

Comment:
I looked at it and decided against implementing something like path. In my opinion it feels like an extended string class with some methods for strings which represents paths and this is something we can fully cover with free functions.

`is_directory` and `is_regular_file` actually determine if the entity exists and if it is a file or directory. Something I would like to add to `cxx::filesystem` at some point. But in this case I look at the string itself and do not care at this stage if it exists or if it is a directory or file. I just would like to analyse the syntax of the string.

But it would make sense to move those free functions in the filesystem header. If you wish I can do this in this PR or in a follow up one.